pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -361,13 +365,39 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(
 	return httpListeners
 }
 
-func buildGatewayListenerTLSContext(server *networking.Server) *auth.DownstreamTlsContext {
+func buildGatewayListenerTLSContext(server *networking.Server, enableSds bool) *auth.DownstreamTlsContext {
 	// Server.TLS cannot be nil or passthrough. But as a safety guard, return nil
 	if server.Tls == nil || model.IsPassThroughServer(server) {
 		return nil // We don't need to setup TLS context for passthrough mode
 	}
 
-	var certValidationContext *auth.CertificateValidationContext
+	tls := &auth.DownstreamTlsContext{
+		CommonTlsContext: &auth.CommonTlsContext{
+			AlpnProtocols: ListenersALPNProtocols,
+		},
+	}
+
+	if enableSds {
+		tls.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{
+			model.ConstructSdsSecretConfigForGatewayListener(server.Hosts[0], model.IngressGatewaySdsUdsPath),
```

Comment:
instead of update existing construct sdsconfig logic to reuse here, I would suggest to write your own function to construct sdsconfig, for ingress case there is no need for envoy to fetch/pass token, so the sdsconfig would be much simpler.

Reply:
Done. Keep current sds config logic unchanged. Move ingress gateway sds config into a separate function.